lib/kubeclient_next/client/api_builder.rb

```diff
@@ -8,7 +8,7 @@
 module KubeclientNext
   module Client
     class APIBuilder < Module
-      attr_reader :api, :client
+      attr_reader :api, :config, :context
```

Comment:
Thought for future:

Instead of abusing `instance_eval`, perhaps a better strategy is to refactor this in a way that we can just include into the `API` class. The trick here, of course, is to separate the abstractions in such a way as to minimize tight coupling. Will need to think more about this, not sure if it's any better or worse than what's currently here.